lib/chef/resource_resolver.rb

```diff
@@ -127,19 +127,38 @@ def self.priority_map
       Chef.resource_priority_map
     end
 
+    def self.handler_map
+      Chef.resource_handler_map
+    end
+
     def priority_map
       Chef.resource_priority_map
     end
 
+    def handler_map
+      Chef.resource_handler_map
+    end
+
     # @api private
     def potential_handlers
-      priority_map.list_handlers(node, resource_name, canonical: canonical)
+      handler_map.list(node, resource_name, canonical: canonical).uniq
     end
 
     def enabled_handlers
       potential_handlers.select { |handler| !overrode_provides?(handler) || handler.provides?(node, resource_name) }
     end
 
+    def prioritized_handlers
+      @prioritized_handlers ||= begin
+        enabled_handlers = self.enabled_handlers
+
+        prioritized = priority_map.list(node, resource_name, canonical: canonical).flatten(1)
+        prioritized &= enabled_handlers # Filter the priority map by the actual enabled handlers
```

Comment:
The issue (and the reason I was being so careful here) is that Ruby's sort is not stable: if there is no priority array, this particular `sort_by` will scramble the array.